src/context/call_reply.rs

```diff
@@ -632,6 +634,8 @@ pub enum CallReply<'root> {
     VerbatimString(VerbatimStringCallReply<'root>),
 }
 
+unsafe impl<'root> Send for CallReply<'root> {}
```

Comment:
The reason it is not safe is because pointer (NonNull) is consider not safe. There is no real reason to wrap the `CallReply` with some Arc, Mutex, or RwLock and we will just lose performance, it is safe to pass the `CallReply` to other threads. I guess that instead of implement `Send` we can use [`unique`](https://docs.rs/unique/latest/unique/struct.Unique.html) but it will give us the same effect...

Reply:
Sorry, we can not user unique because it is still an experiment API.

Reply:
> This is precisely why it is unsafe to pass to other threads: because the NonNull isn't safe, just as you said:

NonNull is consider unsafe because rust do not know if the data behind it is safe to pass between threads of not. This is why it explicitly want us to tell him by implementing the unsafe Send trait.

> As the probability is low that we violate this right now, I think we should comment within the code as to why we implement this trait instead of relying on it being auto-implemented, and why it won't be violated if we implement it on our own.

There is nothing on the CallReply which is thread specific and deny it from been pass to another thread. I will add a comment...